pkg/resourcetopologyexporter/resourcetopologyexporter.go

```diff
@@ -156,9 +170,21 @@ func NewResourceMonitor(cli podresourcesapi.PodResourcesListerClient, args resou
 	}, nil
 }
 
-func (rm *ResourceMonitor) Run(eventsChan <-chan PollTrigger) (<-chan nrtupdater.MonitorInfo, chan<- struct{}) {
+func NewCondition(status v1.ConditionStatus) v1.PodCondition {
```

Comment:
we have the condition types already explicitely listed in the `podreadiness` package, so let's move this (and its sibling function in `nrtupdater`) there as well. Once we did this move, we would probably inline `NewConditionTemplate` in the calling sites. Like:
```
package podreadiness

import (
	"time"

	v1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

// RTEConditionType is a valid value for PodCondition.Type
type RTEConditionType string

// These are valid conditions of RTE pod.
const (
	// ResourcesScanned means that resources scanned successfully.
	ResourcesScanned RTEConditionType = "PodresourcesFetched"
	// TopologyUpdated means that noderesourcetopology objects updated successfully.
	TopologyUpdated RTEConditionType = "NodeTopologyUpdated"
)

func NewNodeTopologyUpdatedCondition(status v1.ConditionStatus) v1.PodCondition {
	cond := newConditionTemplate(TopologyUpdated, status)
	if status == v1.ConditionFalse {
		cond.Reason = "UpdateFailed"
		cond.Message = "failed to update noderesourcetopology object"
	}
	return cond
}

func NewPodresourcesFetchedCondition(status v1.ConditionStatus) v1.PodCondition {
	cond := newConditionTemplate(ResourcesScanned, status)

	if status == v1.ConditionFalse {
		cond.Reason = "ScanFailed"
		cond.Message = "failed to scan pod resources"
	}

	return cond
}

func newConditionTemplate(condType RTEConditionType, status v1.ConditionStatus) (condition v1.PodCondition) {
	return v1.PodCondition{
		Type:               v1.PodConditionType(condType),
		Status:             status,
		LastTransitionTime: metav1.Time{Time: time.Now()},
	}
}
```

Reply:
We don't need the ```NewPodresourcesFetchedCondition``` and ```NewNodeTopologyUpdatedCondition``` if we're using your previous suggestion:
https://github.com/k8stopologyawareschedwg/resource-topology-exporter/pull/65#discussion_r751100455
Because we're updating the ```Reason``` and ```Message``` in the switch case.
But the general suggestion of moving everything into the ```podreadiness``` sounds fair to me.

Reply:
this comment is indeed obsoleted by https://github.com/k8stopologyawareschedwg/resource-topology-exporter/pull/65#discussion_r751100455

Reply:
**Deleted the comment by mistake:** We don't need the NewPodresourcesFetchedCondition and NewNodeTopologyUpdatedCondition if we're using your previous suggestion:
#65 (comment)
Because we're updating the Reason and Message in the switch case.
But the general suggestion of moving everything into the podreadiness sounds fair to me.